src/roster/signals.py

```diff
@@ -120,15 +120,14 @@ def update_social_email(sender, **kwargs):
         # No other User has `social_email` as their email -- good.
         logger.debug('While checking update email for %s, no other Clusive user has that email', social_email)
 
-    # Update
+    # Update clusive user record's email
     if clusive_user.user.email.lower() != social_email:
         clusive_user.user.email = social_email
         clusive_user.user.save()
 
-    try:
-        email_address = EmailAddress.objects.get(user_id=clusive_user.user.id, primary=True)
-        if email_address.email.lower() != social_email:
+    # Update only the primary email address record for this clusive user
+    email_addresses = EmailAddress.objects.filter(user_id=clusive_user.user.id)
+    for email_address in email_addresses:
+        if email_address.primary and email_address.email.lower() != social_email:
```

Comment:
I don't understand the reason for this change. I think the 'try' was put in to avoid errors that actually happened, and the more specific query should be more efficient.

Reply:
Thanks for catching this.  The reason was that the exception actually occurred  -- `get()` found two records.  The reason for that was my Bookshare provider class generated a Bookshare `EmailAddress` record with its `primary` value set to `True`. As far as the database is concerned, there is no contradiction in having multiple primary `EmailAddress`es for the same user.  The fix was (1) to change the `EmailAddress` generation to mark a Bookshare email address as not primary, and (2) to change the logic here in `signals.py` so that even if there are multiple `primary` `EmailAddress` records, then `filter()` will not throw, and only the primary ones will be updated.

But, at the moment, we only support a Bookshare connection involving a non-primary email.  The second change is overkill in that case.  I'll revert back to the former `try: get() ... except: ...` logic.